src/main/java/org/opentripplanner/middleware/triptracker/TripStatus.java

```diff
@@ -66,33 +66,34 @@ public enum TripStatus {
      */
     public static TripStatus getTripStatus(TravelerPosition travelerPosition) {
         if (travelerPosition.expectedLeg != null) {
-            if (travelerPosition.legSegmentFromTime != null &&
-                isWithinModeBoundary(travelerPosition.currentPosition, travelerPosition.legSegmentFromTime)
-            ) {
-                return TripStatus.ON_SCHEDULE;
-            }
             if (travelerPosition.legSegmentFromPosition != null &&
                 isWithinModeBoundary(travelerPosition.currentPosition, travelerPosition.legSegmentFromPosition)
             ) {
-                Instant segmentCenterTime = travelerPosition
+                Instant segmentStartTime = travelerPosition
                     .expectedLeg
                     .startTime
                     .toInstant()
-                    .plusSeconds((long) getSegmentTimeInterval(travelerPosition.legSegmentFromPosition));
-                return travelerPosition.currentTime.isBefore(segmentCenterTime)
-                    ? TripStatus.AHEAD_OF_SCHEDULE
-                    : TripStatus.BEHIND_SCHEDULE;
+                    .plusSeconds((long) getSegmentStartTime(travelerPosition.legSegmentFromPosition));
```

Comment:
What was the reason to change from `getSegmentTimeInterval` to `getSegmentStartTime`? This change seems to report me "ahead of schedule" even if I am behind schedule.

Reply:
Oh I see, never mind. If I am way behind schedule, the trip monitoring might have switched to the next available trip the next day already...

Reply:
Just to add this. The method now has to cover on schedule so it needs to consider the start and end segment times.